joint_state_publisher/joint_state_publisher/joint_state_publisher

```diff
@@ -216,11 +216,18 @@ class JointStatePublisher():
                     parent = param['parent']
                     factor = param.get('factor', 1)
                     offset = param.get('offset', 0)
+                    # Handle recursive mimic chain
+                    recursive_mimic_chain_joints = [name]
                     while parent in self.dependent_joints:
+                        if parent in recursive_mimic_chain_joints:
+                            error_message = "Found an infinite recursive mimic chain"
+                            rospy.logerr("%s: [%s, %s]", error_message, ', '.join(recursive_mimic_chain_joints), parent)
+                            sys.exit(-1)
+                        recursive_mimic_chain_joints.append(parent)
                         param = self.dependent_joints[parent]
                         parent = param['parent']
                         factor *= param.get('factor', 1)
-                        # the offset is relative only to the first parent
+                        offset = factor*offset + param.get('offset', 0)
```

Comment:
Your code does not match your math.

Here is your math:
>    factor = factor_new * factor_old
>    offset = factor_new * offset_old + offset_new

Here is what I think your code does:
>    factor = factor_new * factor_old
>    offset = factor_new * factor_old * offset_old + offset_new

I also think you're looking at the mimic chain from the wrong side. (hence the math is wrong).

Let consider the former example again:
```
 <joint name="Jc" type="continuous">
 </joint>
 <joint name="Jb" type="continuous">
     <mimic joint="Jc" multiplier="x" offset="y"/>
 </joint>
 <joint name="Ja" type="continuous">
    <mimic joint="Jb" multiplier="x'" offset="y'"/>
 </joint>
```
Let do the algorithm step by step, with my modification.
At first you have

name == Ja
factor = x'
offset = y'
parent = Jb

parent in self.dependent_joints == True
parent = Jc
offset += factor * y
offset == y' + x'*y
factor *=  x
factor == x'*x

parent in self.dependent_joints == False

hence:
offset == y' + x'*y
factor == x'*x

we're back on equation (3).